### PR TITLE
fix/syslog-segfault

### LINKED ISSUE #2603: Sys::Syslog is not thread safe can cause segfaults in multithreaded environments

### PR DESCRIPTION
**Pull request by jrouzierinverse**
Description
===========
A Temporary fix for issues with syslog in multithread environments

Impacts
=======
Freeradius and pfsetvlan logging

Issue
=====
fixes #2603 

Delete branch after merge
=========================
NO

NEWS file entries
=================

Bug Fixes
------------

* Fixes segfaults when logging in the multithread environments (#2603)

### CODE REVIEW

File: conf/log.conf.d/pfsetvlan.conf.example [hunk 1]
```diff
@@ -6,8 +6,15 @@ log4perl.rootLogger = INFO, PFSETVLAN
 #log4perl.category.pf.SNMP = WARN
 
 ### Logging for pfsetvlan
-log4perl.appender.PFSETVLAN                                = Log::Dispatch::Syslog
-log4perl.appender.PFSETVLAN.ident                          = packetfence
-log4perl.appender.PFSETVLAN.facility                       = local5
-log4perl.appender.PFSETVLAN.layout.ConversionPattern       = %X{proc}(%X{tid}) %p: [mac:%X{mac}] %m (%M)%n
-log4perl.appender.PFSETVLAN.layout                         = PatternLayout
+log4perl.appender.PFSETVLAN                              = pf::log::FileAppender
+# Replace /usr/local/pf/logs/packetfence.log by /usr/local/pf/logs/pfsetvlan.log to allow
```
Comment (julsemaan) [inline conf/log.conf.d/pfsetvlan.conf.example:10]: Either the comment of the parameter below is wrong

Comment (jrouzierinverse) [reply conf/log.conf.d/pfsetvlan.conf.example:10]: I am not sure what you want done it sound like you gave up mid sentence.

Comment (julsemaan) [reply conf/log.conf.d/pfsetvlan.conf.example:10]: s/of/or/

And anyway my comment was wrong

**Comment by dwlfrth** [comment]
As discussed, this will fix the syslog segfault issue for multithread environments while waiting for the Log::Dispatch upgrade (#2601).
This will address the issue as we know it from 7.3 +
For earlier versions of PacketFence, a maintenance patch will be supplied to fix the bogus "packetfence-multi-domain" FreeRADIUS module (@dwlfrth).

**Comment by dwlfrth** [comment]
Took care of the maintenance patch (#2607)

File: conf/log.conf.d/rlm_perl.conf.example [hunk 2]
```diff
@@ -0,0 +1,18 @@
+### pfsetvlan logger ###
```
Comment (fdurand) [inline conf/log.conf.d/rlm_perl.conf.example]: rlm_perl logger

File: conf/log.conf.d/pfsetvlan.conf.example [hunk 3]
```diff
@@ -6,8 +6,15 @@ log4perl.rootLogger = INFO, PFSETVLAN
...
+# pfsetvlan to log in its own log file.
+log4perl.appender.PFSETVLAN.filename                     = /usr/local/pf/logs/packetfence.log
+log4perl.appender.PFSETVLAN.syswrite                     = 1
+log4perl.appender.PFSETVLAN.mode                         = append
+log4perl.appender.PFSETVLAN.layout                       = PatternLayout
+log4perl.appender.PFSETVLAN.layout.ConversionPattern     = %d{MMM dd HH:mm:ss} %X{proc}(%X{tid}) %p: %m (%M)%n
+log4perl.appender.PFSETVLAN.umask                        = 0002
+log4perl.appender.PFSETVLAN.user                         = pf
+log4perl.appender.PFSETVLAN.group                        = pf
+log4perl.appender.PFSETVALN.binmode                      = utf8
```
Comment (fdurand) [inline conf/log.conf.d/pfsetvlan.conf.example]: PFSETVLAN

File: conf/log.conf.d/rlm_perl.conf.example [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+log4perl.appender.RLM_PERL                              = pf::log::FileAppender
+log4perl.appender.RLM_PERL.filename                     = /usr/local/pf/logs/packetfence.log
+log4perl.appender.RLM_PERL.syswrite                     = 1
+log4perl.appender.RLM_PERL.mode                         = append
+log4perl.appender.RLM_PERL.layout                       = PatternLayout
+log4perl.appender.RLM_PERL.layout.ConversionPattern     = %d{MMM dd HH:mm:ss} %X{proc}(%X{tid}) %p: %m (%M)%n
+log4perl.appender.RLM_PERL.umask                        = 0002
+log4perl.appender.RLM_PERL.user                         = pf
+log4perl.appender.RLM_PERL.group                        = pf
+log4perl.appender.PFSETVALN.binmode                      = utf8
```
Comment (fdurand) [inline conf/log.conf.d/rlm_perl.conf.example]: PFSETVLAN ?

**Comment by jrouzierinverse** [comment]
BUMP

**Pull request merged**